plugins/woocommerce-blocks/assets/js/base/components/button/index.tsx

```diff
@@ -23,48 +31,77 @@ export interface ButtonProps
 	 * Button variant
 	 */
 	variant?: 'text' | 'contained' | 'outlined';
-	/**
-	 * The URL the button should link to.
-	 */
-	href?: string | undefined;
-}
 
-export interface AnchorProps extends Omit< ButtonProps, 'href' > {
-	/**
-	 * Button href
-	 */
-	href?: string | undefined;
+	removeTextWrap?: boolean;
 }
 
 /**
  * Component that visually renders a button but semantically might be `<button>` or `<a>` depending
  * on the props.
  */
-const Button = ( {
-	className,
-	showSpinner = false,
-	children,
-	variant = 'contained',
-	...props
-}: ButtonProps ): JSX.Element => {
-	const buttonClassName = classNames(
-		'wc-block-components-button',
-		'wp-element-button',
-		className,
-		variant,
-		{
-			'wc-block-components-button--loading': showSpinner,
+const Button = forwardRef< HTMLButtonElement, ButtonProps | LinkProps >(
+	( props, ref ) => {
+		const {
+			className,
+			showSpinner = false,
+			children,
+			variant = 'contained',
+			// To maintain backward compat we render a wrapper for button text by default,
+			// but you can opt in to removing it by setting removeTextWrap to true.
+			removeTextWrap = false,
```

Comment:
I think in general it would be nicer for consumers of Button to do things like adding a text wrapper, but for simplicity of migration to the new component I opted to leave things as they were before. I just need to opt out of rendering the wrapper for the buttons used in the radio button scenario.